inter/block.go

```diff
@@ -1,45 +1,188 @@
 package inter
 
 import (
-	"github.com/Fantom-foundation/lachesis-base/hash"
+	"math/big"
+	"unsafe"
+
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
 )
 
+// Block represents the on-disk storage format of a block. It contains all
+// fields required to reconstruct the block header, as well as a list of
+// hashes of the transactions been executed as part of the represented block.
+//
+// This struct should be considered immutable. No fields should be modified,
+// directly or indirectly. Ideally, all fields should be private, but that
+// would invalidate support for RLP encoding as it is used to store instances
+// on the disk. However, future updates may make fields inaccessible.
+//
+// To create a new block, use the BlockBuilder, handling the computation of
+// key properties implicitly.
 type Block struct {
-	Time        Timestamp
-	Atropos     hash.Event
-	Events      hash.Events
-	Txs         []common.Hash // non event txs (received via genesis or LLR)
-	InternalTxs []common.Hash // DEPRECATED in favor of using only Txs fields and method internal.IsInternal
-	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
-	GasUsed     uint64
-	Root        hash.Hash
+	// Fields required for the block header.
+	Number               uint64
+	ParentHash           common.Hash
+	StateRoot            common.Hash
+	Time                 Timestamp
+	Difficulty           uint64
+	GasLimit             uint64
+	GasUsed              uint64
+	BaseFee              *big.Int
+	PrevRandao           common.Hash
+	TransactionsHashRoot common.Hash
+	ReceiptsHashRoot     common.Hash
+	LogBloom             types.Bloom
+
+	// Fields required for linking blocks to contained transactions.
+	TransactionHashes []common.Hash
+
+	// Fields required for linking the block internally to a lachesis epoch.
+	Epoch idx.Epoch
+
+	// The hash of this block, cached on first access.
+	hash common.Hash
+}
+
+// Hash computes the hash of this block, committing all its fields.
+func (b *Block) Hash() common.Hash {
+	if b.hash == (common.Hash{}) {
+		b.hash = b.GetEthereumHeader().Hash()
+	}
+	return b.hash
+}
+
+// uncleHash is the hash to be used for the uncle field in Ethereum headers if
+// there are no uncles. See https://eips.ethereum.org/EIPS/eip-4844.
+var uncleHash = common.BytesToHash(hexutil.MustDecode(
+	"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+))
```

Comment:
You can use `types.EmptyUncleHash` defined in go-ethereum. (the dependency is already here through the `types.Header` type)